### PR TITLE
Tor patch

### PR DESCRIPTION
**Pull request by nyxnor**
#### ISSUE

* https://github.com/rootzoll/raspiblitz/issues/2409

#### BRANCH

* [tor-patch](https://github.com/nyxnor/raspiblitz/tree/tor-patch)

### ISSUES TO CLOSE
* https://github.com/rootzoll/raspiblitz/issues/2054
* https://github.com/rootzoll/raspiblitz/issues/2020
* https://github.com/rootzoll/raspiblitz/issues/2306
* https://github.com/rootzoll/raspiblitz/issues/2273 
* https://github.com/rootzoll/raspiblitz/issues/2395
* https://github.com/rootzoll/raspiblitz/issues/2383
* https://github.com/rootzoll/raspiblitz/issues/2357
* https://github.com/rootzoll/raspiblitz/issues/2069
* https://github.com/rootzoll/raspiblitz/issues/2013
* https://github.com/rootzoll/raspiblitz/issues/1752
* https://github.com/rootzoll/raspiblitz/issues/1604
* https://github.com/rootzoll/raspiblitz/issues/2209 partially
* https://github.com/rootzoll/raspiblitz/issues/2217 partially

### TO DO
- [x] Need to improve tor.update.sh and include that on Tor update options on the updateMenu.
- [x] Search keyword TODO in the files.
- [x] Small note in the FAQ mentioning how to connect to Tor network under a censorship environment.
- [x] Remove tor package form sd card to have a minimalist build and place somewhere else I need to figure where
- [x] Use TPO hidden service sources by default
- [ ] Keep in sync with the next TorBox release v0.4.2 - Release 02 of August of 2021

#### IMPROVEMENTS

Some items could be classified in more than one category, divided here by the most heavy characteristic they hold and expected to provide.

* GENERAL IMPROVEMENT
  * [More Tor uptime (reload instead of restart)](https://github.com/nyxnor/raspiblitz/blob/35a2630e5db438064bdc7f7d1be64e8271931b63/home.admin/config.scripts/tor.functions.lib#L166)
  * [Informational texts of what each tool does](https://github.com/nyxnor/raspiblitz/tree/tor-patch/home.admin/text)
  * [See onion services credentials](https://github.com/nyxnor/raspiblitz/blob/35a2630e5db438064bdc7f7d1be64e8271931b63/home.admin/config.scripts/tor.onion-service.sh#L465)
    * scan QR code when using it on Tor Browser mobile.
    * Output text string on terminal
  * [Typo TOR to Tor](https://github.com/rootzoll/raspiblitz/pull/2419/commits/ca0e5267d797e082c8c091906b0c0f2e585f6d90)
  * [General files permissions to be executable](https://github.com/rootzoll/raspiblitz/pull/2419/commits/78b87d36110ddc8b346b3863429457e714cc0aec)
  * Build under a heavy censorship environment
  * [Maintain ssh hidden service](https://github.com/nyxnor/raspiblitz/blob/35a2630e5db438064bdc7f7d1be64e8271931b63/home.admin/96torMainMenu.sh#L234) (if done through the menu), recommending client auth
  * deprecate instance tor@lnd
  * [Update tor in various ways](https://github.com/nyxnor/raspiblitz/blob/tor-patch/home.admin/config.scripts/tor.update.sh)
    * git and build (git will not ping TPO domain, will use github for that instead of gitlab) - SOURCE CODE
    * apt source (build with dpkg -i) BINARIES
    * apt update (install) - BINARIES

* CIRCUMVENTION
  * [Bridges with Pluggable transport](https://tb-manual.torproject.org/circumvention/) - Bypass censorship with advanced tools
    * [obfs4](https://github.com/nyxnor/raspiblitz/blob/tor-patch/home.admin/config.scripts/tor.bridges-obfs4-activate.sh)
    * [meek](https://github.com/nyxnor/raspiblitz/blob/tor-patch/home.admin/config.scripts/tor.bridges-meek-azure.sh)
    * [snowflake](https://github.com/nyxnor/raspiblitz/blob/tor-patch/home.admin/config.scripts/tor.bridges-snowflake.sh)
  * [APT over Tor](https://github.com/nyxnor/raspiblitz/blob/35a2630e5db438064bdc7f7d1be64e8271931b63/home.admin/96torMainMenu.sh#L296)
    * Debian source
    * TPO sources
  * [Bypass idle](https://github.com/nyxnor/raspiblitz/blob/35a2630e5db438064bdc7f7d1be64e8271931b63/home.admin/96torMainMenu.sh#L381)
    * can use this on airports
    * restaurants
  * [Tor over VPN](https://github.com/nyxnor/raspiblitz/blob/35a2630e5db438064bdc7f7d1be64e8271931b63/home.admin/96torMainMenu.sh#L354)
    * if someone wants with its appropriate warnings
    * Not anonymous depending on how you pay it
    * Not an advanced circumvention tool native to Tor 
  
* SECURITY
  * [Vanguards ](https://github.com/nyxnor/raspiblitz/blob/35a2630e5db438064bdc7f7d1be64e8271931b63/home.admin/config.scripts/tor.onion-service.sh#L310)- Designed to protect onion services that requires high security - [official REPO](https://github.com/mikeperry-tor/vanguards)
    * protection against DDoS
    * against server location deanonymization
  * [Client authorization](https://github.com/nyxnor/raspiblitz/blob/35a2630e5db438064bdc7f7d1be64e8271931b63/home.admin/config.scripts/tor.onion-service.sh#L177)  - TPO [guide](https://community.torproject.org/onion-services/advanced/client-auth/)
    * that requires and extra key when using a onion service
    * can be pasted in the browser window or inserted in torrc for services such as SSH over Tor
  * [Renew addresses](https://github.com/nyxnor/raspiblitz/blob/35a2630e5db438064bdc7f7d1be64e8271931b63/home.admin/config.scripts/tor.onion-service.sh#L292)
    * You can now renew any onion address
    * or bulk renew selected ones to change their onion address
    * it will also change their client authorization key.

* ANONYMIZATION
  * [Change circuit](https://github.com/nyxnor/raspiblitz/blob/35a2630e5db438064bdc7f7d1be64e8271931b63/home.admin/96torMainMenu.sh#L247) - renew circuits if you want to make a connection on a different path
    * When broadcasting tx
    * making new connections to could be potentially faster 
  * [Change entry node](https://github.com/nyxnor/raspiblitz/blob/35a2630e5db438064bdc7f7d1be64e8271931b63/home.admin/96torMainMenu.sh#L253)
    * they are fixed for one month, now you can choose when to change (overusing this won't help you)

### REVIEWERS

* Menu files are in the home directory named 96tor*
* Scripts are in config.scripts/tor.*
* The order of the commands matter.
* Even though it seems a big PR, it is easy to review, clean code and well structured.
* Feel free to code to help, it means a lot. See the tasks above for what remains to be done. Updating tor options and organizing it on the the updateMenu would help a lot.
* Even though bridges scripts were sourced from TorBox, they do not need to be updated frequently, it is just good to get the latest changes every Blitz update to have a better working system and new features.
    
### THANKING

* @radio24 for the project [TorBox](https://github.com/radio24/TorBox) and contributors such as Patrick Truffer, Carlos Martinez for the bridges scripts, over vpn, bypass idle, change entry node, trap script. PS: still have to contributing some enhancements back after more testing.
* mikeperry for answering my questions about Vanguards
* the reviewers

### CODE REVIEW

**Comment by openoms** [comment]
Now I made the PR `Ready for review`, but it exposed a lot of conflicts. @nyxnor can you look at them? Hope they just minor things.

File: home.admin/99connectMenu.sh [hunk 1]
```diff
@@ -257,7 +257,7 @@ HiddenServicePort 8333 127.0.0.1:8333" | sudo tee -a /etc/tor/torrc
         clear
         echo "# Make sure the bitcoind wallet is on"
         /home/admin/config.scripts/network.wallet.sh on
-        /home/admin/config.scripts/internet.hiddenservice.sh bitcoin${BITCOINRPCPORT} ${BITCOINRPCPORT} ${BITCOINRPCPORT}
+        /home/admin/config.scripts/tor.onion-servce.sh bitcoin${BITCOINRPCPORT} ${BITCOINRPCPORT} ${BITCOINRPCPORT}
```
Comment (openoms) [inline home.admin/99connectMenu.sh]: typo, should be:
```
 /home/admin/config.scripts/tor.onion-service.sh bitcoin${BITCOINRPCPORT} ${BITCOINRPCPORT} ${BITCOINRPCPORT}
 ```

File: home.admin/config.scripts/tor.onion-service.sh [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+    exit 0
+  fi
+
+  # This can be used for updates, as there is no official release tag since April of 2019. We are in 2021
+  # https://github.com/mikeperry-tor/vanguards/releases
+  if [ "${STATUS}" == "install" ]; then
+    sudo rm -rf ${DATA_DIR}/vanguards
+    echo "Installing necessary packages..."
+    # installing from git to retrieve the latest version (debian package is not updated frequently)
+    sudo git clone https://github.com/mikeperry-tor/vanguards
```
Comment (openoms) [inline home.admin/config.scripts/tor.onion-service.sh]: Here, but generally everywhere you can avoid using the root user by running the commands with:
`sudo -u debian-tor ...`
Might be relevant in most cases where `set_owner_permission` is used.


To pin the version to a commit :
```
sudo -u debian-tor git clone https://github.com/mikeperry-tor/vanguards
cd vanguards || exit 1
# latest commit in https://github.com/mikeperry-tor/vanguards
sudo -u debian tor reset --hard COMMITHASH || exit 1
```

**Comment by nyxnor** [comment]
> Now I made the PR `Ready for review`, but it exposed a lot of conflicts. @nyxnor can you look at them? Hope they just minor things.

It is from the wrong commit https://github.com/rootzoll/raspiblitz/commit/1f11960cb4e7d759d469d88589656a3783c9eb4b
If this one is removed, it will solve 95%.
How can I do that?

**Comment by openoms** [comment]
it is tricky, I try:

I would fork the original branch you are PR-ing from.
then reset the original branch to before the unwanted commit.
then cherry pick the wanted commits from the forked branch to the original branch
https://git-scm.com/docs/git-cherry-pick
force push

Alternatively :
fork the original branch you are PR-ing from.
reset the forked branch to before the unwanted commit.
apply only the wanted changes (this would practically squash all the commits afterwards)
force push
then merge it to the original branch.

EDIT: 
or even better you might be able to sort it just with:
`git revert 1f11960`
and push

**Comment by openoms** [comment]
Now you can rebase  or merge the upstream `pre-1.7.1` branch to your original PR branch to resolve the conflicts.
Will need to clean all conflicting files one by one to finish the merge.

**Comment by nyxnor** [comment]
@openoms conflicts solved.
@rootzoll `/home/admin/_tor.commands.sh` was previously called `/home/admin/config.scripts/tor.functions.lib` https://github.com/rootzoll/raspiblitz/pull/2419/commits/cc4272a29bf836ec66f4fedc3c94ef6ea8431059. I changed that because not every system has a `config.scripts` folder in the user dir. Want me to revert that?

For both of you:
I set the onion services dir to be `/mnt/hdd/tor/services/electrs` because in the future:
*  it can loop inside that folder too read what service is active. It will make the scripts smaller, easier to review, avoiding typos.
* If I loop inside that folder, the is no need to check if `ElectRS=on` in `raspiblitz.conf`, if the folder is there, it is `on`, cause I set to delete the specific service dir if the service if deactivate/removed (set to off).
* This makes adding any new onion service like watchtower, that is not a Blitz menu option, appear on the Tor menu to be managed, as well as any other services.

**Comment by openoms** [comment]
@nyxnor as we dicussed on the call: on the raspiblitz there is always a `config.scripts` directory.
The lib file makes more sense since it is not containing shortcut commands for the user. Since it used by the scripts, not user-facing makes sense to keep it in config.scripts.

Regarding the Tor services: they are only active if configured in the `torrc`.  A directory being there does not mean it is being served.
But I agree that from the point of view of Tor looking in the torrc might make more sense than in raspiblitz.conf. 

Not sure why the path need to be extended to  `/mnt/hdd/tor/services/SERVICENAME`, so far it was just `/mnt/hdd/tor/SERVICENAME`. Let's keep it like that so there is no need to move existing files.

**Comment by nyxnor** [comment]
I transformed `internet.tor.sh` into `tor.install.sh`, but it includes tor and bitcoin configuration, what would be better is to separate this.
* `tor.install.sh`
* `tor.remove.sh`
* `tor.update.sh`
* `bitcoin.install.sh`
* `bitcoin.update.sh`

When setting run behind Tor, calls `tor.install.sh` then call `bitcoin.install.sh config-tor`.

**Comment by openoms** [comment]
install, remove, update and config can be in the same script.
Bitcoin is installed in the `build_sdcard.sh` and has a `bitcoin.chains.sh` for parallel services (could have a better name) and there is also a `bitcoin.update.sh` for updates.
They all should just call the `tor.install.sh` accordingly.

**Comment by nyxnor** [comment]
Last phase, what remains to be done is listed on the task of the first comment of this PR, which is:
1. finish `tor.update.sh`

Done for this PR.

2. sync (manually) with TorBox commits (wait version v0.4.2 coming in the beginning of august)
For syncing with TorBox, always write the last commit reviewed to [`tor.functions.lib`](https://github.com/nyxnor/raspiblitz/blob/823e3a3ad9a68dbd57913022f2624a1b2775987e/home.admin/config.scripts/tor.functions.lib#L35).

This remains to be done.

**Comment by nyxnor** [comment]
### ATTENTION

This is an important topic to discuss, this commit https://github.com/rootzoll/raspiblitz/pull/2419/commits/76f46c09d92d1c339427a691ddd778de4f04e8f9

Bridges with Pluggable Transport are Obfs4, Meek and Snowflake.
Let's discard meek and snowflake from the discussion because they work differently and that is their only ip.

Point is obfs4 bridges lines being listed in the repository.
What happens is that a censor can try to block those bridges and bridges you use should not be made available when you are connecting to it at least sigh.

But being listed there does not mean you will use all those bridges. You just can use 3 bridges at a time, there are 36 listed, meaning there is an 0,08333 chance of using that list, but and a big interjection:
* this probability is raw, not fine tuned to all metrics. Probability is lower than that if digging deeper
* without trying to use an _argumentum ad vericundiam_, TorBox chose this way for their [torrc](https://raw.githubusercontent.com/radio24/TorBox/master/etc/tor/torrc) too to facilitate usage, but they probably though of all the points below before me
* you are not obliged to use that bridges, they can be activated and deactivated at will
* you can add your own private bridges
* add bridges
  * manually
  * from the TPO database
* edit your local git repo and change to different bridges
* does not mean you will use more than one bridge either, you can activate just one bridge

Time correlation attacks?
Not all Blitz users will use bridges, there is thorough explanation that is should only be enabled by those who need it.
Not all those bridge users are Blitz users.

### Note: No tor sensitive logs are shared public in the PR, which is very important for privacy and security
Sharing the log would show which time and what bridge you were using at that moment.

**Comment by xanoni** [comment]
_**Topic 1/2 — Mosh to Mask SSH Delays**_

What's your perspective on `mosh` (https://mosh.org/)? When I `ssh` into my Pi Hidden Service, the keystrokes are terribly delayed (as expected when there are 6 hops).

Mosh is really good at masking such delays and makes `ssh` feel more natural when pings are high. Some people voiced concerns about its partially self-rolled encryption, but that shouldn't be a concern if we have Tor encryption as well.

While `mosh` is compatible with `sshd`, it needs its own separate UDP side-channel, and `tor` can only talk TCP. I'm trying to figure out if there's a solution by now, see https://github.com/mobile-shell/mosh/issues/13


_**Topic 2/2 — Lower Latency Tor Circuits**_

Another potential approach, and I really don't like it that much and don't know how much it helps, would be restricting the Tor circuits used for `ssh` to specific geographically close nodes ... see `EntryNodes`, `MiddleNodes`, `ExitNodes` here: https://2019.www.torproject.org/docs/tor-manual.html.en

Related issue: https://github.com/rootzoll/raspiblitz/issues/2273

**Comment by nyxnor** [comment]
@xanoni 
1 - Mosh -> Never heard about it, seems interesting but it is client side, not server side.
2 - Defining geographically close nodes -> That is not wise privacy wise. Latency is not high because of distant nodes, it is high because of the load passing through those nodes. Also, I cannot choose users hops, this is advanced and should be used only by users who knows how to configure it.

**Comment by xanoni** [comment]
> @xanoni 
> 
> 1 - Mosh -> Never heard about it, seems interesting but it is client side, not server side.

Actually it is both ... you need it on the client and on the server (but it's the same apt package). But if you look at my linked thread, it seems that tunneling the UDP control stream over Tor could be a challenge.

> 2 - Defining geographically close nodes -> That is not wise privacy wise. Latency is not high because of distant nodes, it is high because of the load passing through those nodes. Also, I cannot choose users hops, this is advanced and should be used only by users who knows how to configure it.

Yes I agree that I would prefer not to do this.

**Comment by xanoni** [comment]
Is I2P on your roadmap? I played a bit with `i2pd` and they natively support UDP. Thus, it's very easy to use `mosh` over I2P :) :) :)

PS: It seems that the Java implementation (`i2p`) does NOT support UDP, so I'm using the C++ implementation (`i2pd`).

EDIT: see relevant documentation here — https://i2pd.readthedocs.io/en/latest/user-guide/tunnels/#udp-tunnels

**Comment by openoms** [comment]
@xanoni there is a relevant issue to discuss I2P: https://github.com/rootzoll/raspiblitz/issues/2413
shared experiences are welcome.

Mosh is good, but needs more setup on the user's computer which we are trying to avoid and starting to move to
 a web interface.
To discuss further it would worth it's own issue which can even have a short guide for advanced users to try.

**Comment by openoms** [comment]
@nyxnor merging this aimed to be happen after a v1.7.1RC1 base image so testing can be a bit more isolated from all the other changes.
Sorry it takes times, but will take another week for me to get back to work.

**Comment by nyxnor** [comment]
@openoms No problem, the last thing I need to do and is gonna take a bit of time is to manually sync with TorBox v.0.4.2.

**Comment by nyxnor** [comment]
I have been working on a single menu for onion-services that will loop services and greatly reduce the number of lines to be reviewed as well to be maintained.

------
 @openoms take a look how it is possible to maximize the tor.onion-service.sh functionality and to used less lines in a menu. Approved to commit this to Blitz to get rid of the 96tor menus? Also I still need to sync with the latest version of TorBox, so pls do not rush to review as a lot of files could change: https://github.com/nyxnor/CLI-onion-services

------

~~Found time do update to TorBox this weekend, wait for signaling review ready.~~
Will get back to this before 1.8 as the milestone changed below.

**Comment by nyxnor** [comment]
@openoms @rootzoll 
1. I have not integrated the lasted torbox bridges mechanism, I am trying to automate the script renaming and changing paths to fit Blitz enviroment so I don't need to do much manual work every update, it is going fine. Script should get the lasted torbox release, copy specified scripts to a new folder, rename scripts name of the files and inside the files, set full path so it can be run from any folder, remove TorBox appearance in the titles but leave the verbatim, change host to localhost instead of what is used there, remove relay scripts and mentions.
2. I will close this PR as it no comments were made, it was rebased to dev, it changed 100 files and I will not edit TOR to Tor to keep it simple, I will just add the core files in the next PR to facilitate review.
3. I did not like how the services were hardcoded in the menu so I made it dynamic and less bloat. https://github.com/nyxnor/onion-cli. Please review that repo or share thoughts because the next PR will greatly use this (but less options as blitz will handle service creation and removal. 
3.1 It does what [auth - 188](https://github.com/nyxnor/raspiblitz/blob/tor-patch/home.admin/96torAuthMenu.sh), [renew - 230 lines](https://github.com/nyxnor/raspiblitz/blob/tor-patch/home.admin/96torRenewAddressMenu.sh), [credentials - 240 lines](https://github.com/nyxnor/raspiblitz/blob/tor-patch/home.admin/96torCredentialsMenu.sh), [onion service script - 500 lines](https://github.com/nyxnor/raspiblitz/blob/tor-patch/home.admin/config.scripts/tor.onion-service.sh). TOTAL LINES = 1158
3.2 but will way less lines: [onion menu - 426 lines](https://github.com/nyxnor/onion-cli/blob/main/onion-menu.sh), [onion script - 760 lines](https://github.com/nyxnor/onion-cli/blob/main/onion-service.sh). TOTAL LINES = 1186.
3.3 But it has 20 more lines.... yes, because it also implements authentication client side, create and export backups, import backup and integrate into the system, create service with unix path, it also checks if the command is valid for every option so it wont run if it is invalid. So it has more functionalities with almost the same amount of lines, less files to manage, no hard coded path or service name, so any new service integrated into blitz will be recognized and there is no need to edit the script.
3.4 the code is working but it is not the most optimized as I would like it to be, but it is this current blitz script https://github.com/rootzoll/raspiblitz/blob/v1.7/home.admin/config.scripts/internet.tor.sh on steroids.

**Read https://github.com/nyxnor/onion-cli/blob/main/TECHNICAL.md**

**Comment by openoms** [comment]
Good to see that you have started a pluggable project for the added Tor functions. It will help to maintain those parts and will make it easier to integrate to other systems.

to further help to make these changes more atomic parts were implemented during development:

Tor is now only reloads not restarts: 4874d7dd
The `tor@lnd` instance will be removed: https://github.com/rootzoll/raspiblitz/pull/2536

User facing capitalization fixes: 66fd41e7

Various other fixes:
47a1407f extend tor status
4c339af3 fix tor detection
69f882fe change tor test
dfaaf992 improve tor detection
32747d40 test tor running

**Comment by nyxnor** [comment]
@openoms I see FreeBSD uses `sh` for the normal user, so aiming to make the script compatible to that as much as possible to it can work on GNU/Linux or BSD facilitates the migration. Unfortunately POSIX compliant does not accept arrays and does not accept sourcing files to get information/variables, so it has its drawbacks. The FreedBSD root shell is tcsh but it won't get invoked, only commands requires sudo, the root user is not used but user with sudo priviles is need. https://docs.freebsd.org/de/articles/linux-users/#shells. My /bin/sh is symlinked to `dash`. Wonder what is symlinked in bsd.

----------
I will be mainly improving the script (CLI) for now, the whiptail/dialog menu (TUI) needs a lot of work to do, it is too slow loading and can be improved. FreeBSD man page for `dialog` is very much in favor of it instead of whiptail, and as dialogs are more customizable and it is what Blitz uses on the main menus, help on the TUI would be appreciated, need to avoid arrays as explained above. https://www.freebsd.org/cgi/man.cgi?query=dialog&sektion=1
I am committing directly to main because I already warned to not trust the repo as it is not rock solid yet and a lot of changes.

**Comment by vv01f** [comment]
@nyxnor 

* array: iterate over (delimited) lines/files
* source: see dot command https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_18_01 

Feel free to throw things that don't seem to work and we'll find some substitute

**Comment by nyxnor** [comment]
@vv01f THANK YOU
Sourcing with dot now.
> array: iterate over (delimited) lines/files

Yeah, need to iterate a folder to get the files name and display them on the TUI. I read that using `ls` is not a good thing to get files names, but the normal user does not have access to the tor user, so I need to use `sudo -u debian-tor ls /var/lib/tor/services/`. If you also have any idea for a better command to do that. Here is an example https://github.com/nyxnor/onionservice/blob/1dd4c7a008f6c2ad47b0a91d86b52784a72f7aec/onionservice-cli#L264. Because just root and tor user have reading and writing access, I can't test with `-f file` or `-d dir` to check if file or directory exists also.

Edit:
dir a gnu util but not compatible with bsd https://github.com/DiegoMagdaleno/BSDCoreUtils/blob/78d57fcc3cd9de3342ed7424b9a78a3dfdc1b8a3/TRADEOFFS#L13
For checking directory, found about the `dir -d` command
```sh
sudo -u debian-tor dir -d /var/lib/tor/services >/dev/null 2>&1 || printf "something else...."
```
And instead of `ls`, I possibly can use `dir` too:
```sh
sudo -u debian-tor dir /var/lib/tor/services
hs_1 hs_2 hs_3
```

**Comment by vv01f** [comment]
@nyxnor 

* iterating over files without `ls`/`dir`: `for file in directory/*; do echo "$file"; done` – also e.g. `*.conf` to match certain subsets.
* for having a certain users/groups rights better run the script with that user/group already instead of switching for single commands; in doubt pull the relevant parts in a script for that user/group

**Comment by nyxnor** [comment]
@vv01f

> for file in directory/*; do echo "$file"; done

That is what I do, but I need to use a command after `sudo -u debian-tor`. See this mentioned above https://github.com/nyxnor/onionservice/blob/1dd4c7a008f6c2ad47b0a91d86b52784a72f7aec/onionservice-cli#L264

**Comment by openoms** [comment]
@nyxnor do you want to try this on top of v1.7.1?

**Comment by nyxnor** [comment]
Yes but wont use this branch as it has been surpassed. Closing now and you will be notified soon.
@openoms

**Comment by nyxnor** [comment]
See https://github.com/rootzoll/raspiblitz/issues/2669

**Pull request closed without merging**